resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/backend/IntegrationTestBackend.scala

```diff
@@ -30,14 +30,16 @@ private[spark] trait IntegrationTestBackend {
 private[spark] object IntegrationTestBackendFactory {
   val deployModeConfigKey = "spark.kubernetes.test.deployMode"
 
+  val backendMinikube = "minikube"
+  val backendDockerForDesktop = "docker-for-desktop"
+
   def getTestBackend: IntegrationTestBackend = {
     val deployMode = Option(System.getProperty(deployModeConfigKey))
-      .getOrElse("minikube")
-    if (deployMode == "minikube") {
-      MinikubeTestBackend
-    } else {
-      throw new IllegalArgumentException(
-        "Invalid " + deployModeConfigKey + ": " + deployMode)
+      .getOrElse(backendMinikube)
+    deployMode match {
+      case `backendMinikube` => MinikubeTestBackend
+      case `backendDockerForDesktop` => DockerForDesktopBackend
```

Comment:
Can you add a README to explain how to run with docker-for-desktop, by specifying --deploy-mode